Makefile

```diff
@@ -80,10 +80,13 @@ help:  ## Display this help
 .PHONY: testprereqs
 testprereqs: $(KUBEBUILDER) $(KUSTOMIZE)
 
-.PHONY: test
-test: testprereqs generate fmt lint ## Run tests
+.PHONY: unit
+unit: testprereqs ## Run tests
 	source ./hack/fetch_ext_bins.sh; fetch_tools; setup_envs; go test -v ./controllers/... ./ipam/... -coverprofile ./cover.out; cd $(APIS_DIR); go test -v ./... -coverprofile ./cover.out
 
+.PHONY: test  ## Run formatter, linter and tests
+test: generate fmt lint unit
```

Comment:
I am wondering if we need `generate` for test target, at least in CAPM3 we are not doing it https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/Makefile#L116-L117. It makes sense to have it though so maybe we should update the CAPM3 makefile "test" target

Reply:
I agree we should add this in CAPM3 if it isn't too heavy to run all the time.